runtime/integration-tests/src/pools/token_prices_normal.rs

```diff
@@ -0,0 +1,209 @@
+		let mut genesis = Storage::default();
+		genesis::default_balances::<Runtime>(&mut genesis);
+		env::test_env_with_centrifuge_storage::<BLOCK_TIME>(&manager, genesis)
+	};
+
+	let mut nft_manager = NftManager::new();
+	let pool_id = 0u64;
+	let loan_amount = 10_000 * DECIMAL_BASE_12;
+	let loan_id = InstanceId(1);
+	let borrow = 9_000 * DECIMAL_BASE_12;
+	let borrow_amount = Amount::from_inner(borrow);
+	let investment = 5_000 * DECIMAL_BASE_12;
+	let maturity = time::date(90 * SECONDS_PER_DAY);
+
+	env::run!(
+		env,
+		Chain::Para(PARA_ID),
+		Call,
+		ChainState::PoolEmpty,
+		Keyring::Admin => default_pool_calls(Keyring::Admin.into(), pool_id, &mut nft_manager),
+			issue_default_loan(
+				Keyring::Admin.into(),
+				pool_id,
+				loan_amount,
+				maturity,
+				&mut nft_manager,
+			)
+	);
+
+	env::assert_events!(
+		env,
+		Chain::Para(PARA_ID),
+		Event,
+		EventRange::Latest,
+		Event::System(frame_system::Event::ExtrinsicFailed{..}) if [count 0],
+		Event::Pools(pallet_pools::Event::Created(id, ..)) if [id == pool_id],
+		Event::Loans(pallet_loans::Event::PoolInitialised(id)) if [id == pool_id],
+		Event::Loans(pallet_loans::Event::Created(id, loan, asset))
+			if [id == pool_id && loan == loan_id && asset == Asset(4294967296, InstanceId(1))],
+		Event::Loans(pallet_loans::Event::Priced(id, loan)) if [id == pool_id && loan == loan_id],
+	);
+
+	env::run!(
+		env,
+		Chain::Para(PARA_ID),
+		Call,
+		ChainState::PoolEmpty,
+		Keyring::TrancheInvestor(0) => invest_order_call(pool_id, 0, investment);
+		Keyring::TrancheInvestor(1) => invest_order_call(pool_id, 0, investment);
+	);
+
+	env::assert_events!(
+		env,
+		Chain::Para(PARA_ID),
+		Event,
+		EventRange::Latest,
+		Event::System(frame_system::Event::ExtrinsicFailed{..}) if [count 0],
+		Event::Pools(pallet_pools::Event::InvestOrderUpdated(id, _tranche, who))
+			if [id == pool_id && who == Keyring::TrancheInvestor(0).to_account_id()],
+		Event::Pools(pallet_pools::Event::InvestOrderUpdated(id, _tranche, who))
+			if [id == pool_id && who == Keyring::TrancheInvestor(1).to_account_id()],
+	);
+
+	env::pass_n(&mut env, 1);
+
+	env::run!(
+		env,
+		Chain::Para(PARA_ID),
+		Call,
+		ChainState::PoolEmpty,
+		Keyring::Admin =>
+			update_nav(pool_id),
+			close_epoch(pool_id),
+			borrow_call(pool_id, loan_id, borrow_amount)
+	);
+
+	env::assert_events!(
+		env,
+		Chain::Para(PARA_ID),
+		Event,
+		EventRange::Latest,
+		Event::System(frame_system::Event::ExtrinsicFailed{..}) if [count 0],
+		Event::Pools(pallet_pools::Event::EpochExecuted(id, ..)) if [id == pool_id],
+		Event::Pools(pallet_pools::Event::EpochClosed(id, ..)) if [id == pool_id],
+		Event::Loans(pallet_loans::Event::Borrowed(id, loan_id, amount))
+			if [id == pool_id && loan_id == loan_id && amount == borrow_amount],
+	);
+
+	env::pass_n(&mut env, 30 * time::blocks_per_day::<BLOCK_TIME>());
+
+	let token_prices = env
+		.with_state(Chain::Para(PARA_ID), || {
+			pools::with_ext::get_tranche_prices(pool_id)
```

Comment:
I didn't understand originally why the token price had already gone up here, but realized later on that you're calling `update_nav` within `get_tranche_prices`. Maybe overly explicit, but wdyt about calling this `update_nav_and_get_tranche_prices`?

Reply:
Yes, agree! will do that.